### PR TITLE
[vcpkg/script] get infos from installed ports

### LINKED ISSUE #8269: [vcpkg] Make information about dependent ports available to currently building port
This is required to correctly detect/transport/setup transitiv dependencies in the current port if it is port which is not build with CMake like Qt.

The following variables would be good: (see also https://github.com/microsoft/vcpkg/pull/8239#issuecomment-533473087)
```
${PORT}_DEPENDENT: Linear unique list of *all* dependent ports
${PORT}_${DEPENDENT}_FEATURES: List of features <dependent> has been built with
${PORT}_${DEPENDENT}_DEPENDENT: List of dependent ports of <dependent>
```

### PR DESCRIPTION
**Pull request by Neumann-A**
some ports require insight into their dependent ports to be configured correctly. One such example is pcl

closes #8269

### CODE REVIEW

File: ports/pcl/portfile.cmake [hunk 1]
```diff
@@ -39,17 +39,20 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     opengl      WITH_OPENGL
     vtk         WITH_VTK
 )
-
+x_vcpkg_get_port_info(PORTS qhull flann boost-system)
+string(COMPARE EQUAL "${boost-system_LIBRARY_LINKAGE}" "dynamic" BOOST_LINKAGE)
```
Comment (Neumann-A) [inline ports/pcl/portfile.cmake:43]: This (boost) needs to be checked by someone.

**Comment by ras0219-msft** [comment]
Usually, this is to set macros for the headers or to change searched-for library names -- in both these cases, we should be solving the problem in the dependency (hardcoding the correct macro setting or using an indirection layer like `pkg-config`/`find_package()`/??? to get the full list of needed libraries, including transitive dependencies). Do you know what PCL uses it for?

**Comment by Neumann-A** [comment]
I just checked the variable for boost and it was setting up the find_package call + adding some compile definitions. 
At least for plugins which don't use cmake there is no way for the consumer to know if it is static or dynamic and thus needs special handling any way.

Also there are more ports than pcl requiring the information since I can remember saying 'this won't work correctly in a custom triplet a few times in different prs'

**Comment by BillyONeal** [comment]
I added requires:discussion back so that we look at Neumann-A's reply to Robert

**Comment by BillyONeal** [comment]
To our understanding, there are 2 expected use cases for this proposal:

* As used here, it's being used to determine which linkage was used in the dependencies, but that should be resolved using pkgconfig or cmake rather than vcpkg making our own convention that must be patched into every port. This may require dependencies to be patched to provide a consistent name that doesn't depend on linkage. See:
  * https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries
  * The "important caveat" at https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-rename-binaries-outside-the-names-given-by-upstream
* For feature detection of dependencies, that should be added as a feature in the consuming port; there should never be a difference in behavior based on features installed in a dependency. Taking advantage of a feature in a dependency should itself be a feature in the dependent which build-depends on the dependency's feature.

We admit that there may be other use cases than enumerated above which might argue to revisit this. However. we believe the above cases are better served as enumerated above.

**Pull request closed without merging**